### PR TITLE
Fixes #37579 - Define correct timezone behavior in unittest

### PR DESCRIPTION
**Pull request by dosas**
see also https://github.com/theforeman/hammer-cli/pull/369#issuecomment-2180564378

### CODE REVIEW

File: test/unit/options/normalizers_test.rb [hunk 1]
```diff
@@ -429,15 +439,45 @@
...
+    it "should parse iso8601 compliant timezones" do
+      mock_env("TZ" => "UTC") do
+        _(formatter.format("1986-01-01 08:30:20 +04:00")).must_equal("1986-01-01T08:30:20+04:00")
+      end
+    end
+
+    it "should not recognize invalid timezones" do
+      mock_env("TZ" => "UTC") do
+        _(formatter.format("1986-01-01 08:30:20 bullshit")).must_equal("1986-01-01T08:30:20+00:00")
+      end
```
Comment (adamruzicka) [inline test/unit/options/normalizers_test.rb:468]: Could you either duplicate this one or change it from utc to something else to be extra clear it's picking up the timezone from the environment and not silently defaulting to utc?

File: test/unit/options/normalizers_test.rb [hunk 2]
```diff
@@ -429,15 +439,45 @@
...
+    end
+
+    it "should parse iso8601 compliant timezones" do
+      mock_env("TZ" => "UTC") do
+        _(formatter.format("1986-01-01 08:30:20 +04:00")).must_equal("1986-01-01T08:30:20+04:00")
+      end
+    end
+
+    it "should not recognize invalid timezones" do
```
Comment (adamruzicka) [inline test/unit/options/normalizers_test.rb:465]: The name is slightly misleading as it also ignores "valid" timezones such as `EDT`.

File: test/unit/options/normalizers_test.rb [hunk 3]
```diff
@@ -429,15 +439,45 @@
...
+      mock_env("TZ" => "UTC") do
+        _(formatter.format("1986-01-01 08:30:20 bullshit")).must_equal("1986-01-01T08:30:20+00:00")
```
Comment (adamruzicka) [inline test/unit/options/normalizers_test.rb:467]: Although I personally don't mind, some might find this invalid time zone inappropriate

**Review by adamruzicka — CHANGES_REQUESTED** [review]

File: test/unit/options/normalizers_test.rb (hunk 2 above)
Comment (dosas) [reply test/unit/options/normalizers_test.rb:465]: What name would you suggest? `should not recognize certain timezones` because it recognises some.

Comment (adamruzicka) [reply test/unit/options/normalizers_test.rb]: maybe something along the lines of `should not recognize timezone strings not formatted according to iso8601`?

Comment (dosas) [reply test/unit/options/normalizers_test.rb]: That is also slightly misleading because it recognizes timezones like `EST`

**Review by adamruzicka — APPROVED** [review]

**Comment by adamruzicka** [comment]
Thank you @dosas !

**Pull request merged**